examples/checkpointing/checkpoint.py

```diff
@@ -17,49 +26,49 @@ def apply(self, t_start, t_end):
             this method passes them on correctly to devito.Operator
         """
         args = self.args.copy()
-        args[self.argnames['t_start']] = t_start
-        args[self.argnames['t_end']] = t_end
+        args[self.t_arg_names['t_start']] = t_start
+        args[self.t_arg_names['t_end']] = t_end
         self.op.apply(**args)
 
 
 class DevitoCheckpoint(Checkpoint):
     """Devito's concrete implementation of the Checkpoint abstract base class provided by
        pyRevolve. Holds a list of symbol objects that hold data.
     """
-    def __init__(self, symbols):
+    def __init__(self, objects):
         """Intialise a checkpoint object. Upon initialisation, a checkpoint
-        stores only a reference to the symbols that are passed into it.
-        The symbols must be passed as a mapping symbolname->symbolobject."""
-        assert(all(isinstance(s, TimeFunction) for s in symbols))
-        self._dtype = symbols[0].dtype
-        self.symbols = symbols
-        self.revolver = None
+        stores only a reference to the objects that are passed into it."""
+        assert(all(isinstance(o, TimeFunction) for o in objects))
+        self.objects = objects
 
     @property
     def dtype(self):
-        return self._dtype
+        try:
+            return self.objects[0].dtype
```

Comment:
What I meant in the original comment was whether we should check that **all** objects/functions have **same** dtype

Reply:
Done